packages/api-v4/src/account/types.ts

```diff
@@ -56,8 +56,24 @@ export interface ActivePromotion {
   this_month_credit_remaining: string;
   credit_monthly_cap: string;
   image_url: string;
+  service_type: string;
 }
 
+export type PromotionServiceType =
+  | 'all'
+  | 'backup'
+  | 'blockstorage'
+  | 'db_mysql'
+  | 'ip_v4'
+  | 'linode'
+  | 'linode_disk'
+  | 'linode_memory'
+  | 'longview'
+  | 'managed'
+  | 'nodebalancer'
+  | 'objectstorage'
+  | 'transfer_tx';
+
```

Comment:
Could this be changed to an Enum instead of a type?
```suggestion
export enum PromotionServiceEnum {
  all = 'All',
  backup = 'Backups',
  blockstorage = 'Volumes',
  dm_mysql = 'DBaas',
  ip_v4 = 'IPv4',
  linode = 'Linodes',
  linode_disk = 'Storage',
  linode_memory = 'Memory',
  longview = 'Longviews',
  managed = 'Managed',
  nodebalancer = 'NodeBalancers',
  objectstorage = 'Object Storage',
  transfer_tx = 'Service Transfers',
};

```
This would then remove the need for the `serviceTypeMap` in `BillingSummary.tsx`

Reply:
This is a cool idea...is there some way to use enums as union types (quick googling says no but I'm not super familiar with these). I'd like to have type safety/autocomplete for the JS client PromoType. Also, the string values are a bit Cloud-specific, so the Enum should probably live outside of api-v4.

Reply:
Do you mean union typing two Enums together or do you mean treating the keys/values as union types?

Reply:
A union type of keys of an enum can be achieved through `keyof typeof PromotionServiceEnum`. the Resulting type can be used to restrict the value assigned to a variable to only the keys of the enum.

Reply:
This seems a bit out of scope for this story as it's a larger architectural change. We can create a tech story to address this if anyone feels strongly.